### ISSUE TITLE
Embedded language detector and overloads

### ISSUE THREAD

**Issue by JamesNK** [dotnet/roslyn#61982]
The embedded language detector doesn't detect a string has `StringSyntax` on it until there is one valid method overload. This is simple and makes sense, but it gives an inconsistent experience. When using an API that has an overload it feels bad that a string isn't being highlighted or completion isn't working, until a certain point of time when you've provided enough arguments that only one method is remaining.

The detector could be improved to check all the overloads, and if the parameter at the position has the same `StringSyntax` value in all overloads then use it.

The reason I noticed it is the popular ASP.NET Core minimal API has two overloads for each method.

```cs
public static class RouteEndpointExtensions
{
    public static IEndpointConvertionBuilder MapGet(this IEndpointRouteBuilder endpoints, [StringSyntax("Route")] string pattern, Delegate delegate);
    public static IEndpointConvertionBuilder MapGet(this IEndpointRouteBuilder endpoints, [StringSyntax("Route")] string pattern, RequestDelegate delegate);
}
```
The pattern parameter isn't highlighted until the overload is figured out, which happens after the pattern and delegate are written.
```cs
endpoints.MapGet("my // <- not detected
endpoints.MapGet("myroute", // <- not detected
endpoints.MapGet("myroute", () => "" // <- finally detected
```
This makes route syntax much less useful. You only get it once you've finished writing the endpoint. It would be better if the embedded language detector could see that both `MapGet` overloads have the same string syntax and use it upfront.

cc @CyrusNajmabadi

**Comment by JamesNK** [comment]
This issue impacts other embedded languages and their APIs. Regex for example:

```cs
Regex.Match("", "$" // <- not detected
Regex.Match("", "$", // <- not detected
Regex.Match("", "$", RegexOptions.IgnoreCase, TimeSpan.FromSeconds(1)); // <- detected
```

**Comment by CyrusNajmabadi** [comment]
Yup.  Should be easy to fix.  One sec :)
